File: Lib/Telerik/Examples/QuickStart/ChartView/LabelFormatting/TwoLabelOhlcSeries.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.Charting;

namespace Telerik.Examples.WinControls.ChartView.LabelFormatting
{
    public class TwoLabelOhlcSeries : OhlcSeries
    {
        protected override DataPointElement CreatePointElement(DataPoint point)
        {
            return new TwoLabelOhlcPointElement(point);
        }
    }

    public class TwoLabelOhlcPointElement : OhlcPointElement
    {
        LabelElement secondLabel;

        public TwoLabelOhlcPointElement(DataPoint point)
            : base(point)
        {
            secondLabel = CreateLabelElement(this);
            this.Children.Add(secondLabel);
        }

        public LabelElement SecondLabel
        {
            get { return this.secondLabel; }
        }

        public override Type GetThemeEffectiveType()
        {
            return typeof(OhlcPointElement);     
        }   
    }
}

File: Lib/Telerik/Examples/QuickStart/ChartView/LabelFormatting/TwoLabelOhlcSeries.vb
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.Charting

Namespace Telerik.Examples.WinControls.ChartView.LabelFormatting
	Public Class TwoLabelOhlcSeries
		Inherits OhlcSeries
		Protected Overrides Function CreatePointElement(ByVal point As DataPoint) As DataPointElement
			Return New TwoLabelOhlcPointElement(point)
		End Function
	End Class

	Public Class TwoLabelOhlcPointElement
		Inherits OhlcPointElement
'INSTANT VB NOTE: The variable secondLabel was renamed since Visual Basic does not allow class members with the same name:
		Private secondLabel_Renamed As LabelElement

		Public Sub New(ByVal point As DataPoint)
			MyBase.New(point)
			secondLabel_Renamed = CreateLabelElement(Me)
			Me.Children.Add(secondLabel_Renamed)
		End Sub

		Public ReadOnly Property SecondLabel() As LabelElement
			Get
				Return Me.secondLabel_Renamed
			End Get
		End Property

		Public Overrides Function GetThemeEffectiveType() As Type
			Return GetType(OhlcPointElement)
		End Function
	End Class
End Namespace
